### PR TITLE
LRQA-66243  Create Automated tests for Clay Card

### PR DESCRIPTION
**Pull request by mateusralv**
Background
Create Automated tests for Clay Card

Test Plan
1. Given Test Page with Clay Sample Portlet added
When view Cards tab
And When viewing default state (not checked) of Selectable Image Card
Then it should not be active state

2. Given Test Page with Clay Sample Portlet added
When view Cards tab
And When click a default state of Selectable Image Card
Then it should check the checkbox
And Then it should make the card active

3. Given Test Page with Clay Sample Portlet added
When view Cards tab
And When click the title of Selectable Image Card
Then it appends the href value to current URL: #image-card-href

4. Given Test Page with Clay Sample Portlet added
When view Cards tab
And When click the elipsis of Selectable Image Card
And When click the Group 1 - Option 1
Then it appends the href value to current URL: # 1

JIRA Ticket:
[LRQA-66243](https://issues.liferay.com/browse/LRQA-66243)

### CODE REVIEW

**Comment by mateusralv** [comment]
ci:test:frontend:nocompile

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/claysample/ClaySamplePortlet.path [hunk 1]
```diff
@@ -36,6 +36,26 @@
...
 	<td></td>
 </tr>
+<tr>
+	<td>CHECKBOX_SELECTABLE_IMAGE_CARD</td>
+	<td>//div[contains(@id,'image-card-icon-block')]//input[contains(@class,'custom-control-input')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>CARD_ACTIVE</td>
+	<td>//div[contains(@class,'active file-card form-check-card form-check form-check-top-left card-type-asset')]</td>
```
Comment (susanchen3) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/claysample/ClaySamplePortlet.path:46]: I don't think we need all the classes. Maybe shorten to: `//div[contains(@class,'active file-card')]`

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/claysample/ClaySamplePortlet.path [hunk 2]
```diff
@@ -36,6 +36,26 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>KEBAB_DROPDOWN_CARD_OPTION</td>
+	<td>//div[contains(@id,'image-card-icon-block')]//button[contains(@class, 'dropdown-toggle component-action btn btn-monospaced btn-unstyled')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>TITLE_CARD</td>
+	<td>xpath=(//div[contains(@class,'autofit-col autofit-col-expand')]//a[contains(text(),'${key_title}')])[${key_index}]</td>
```
[the remark this answers is not in the document]
Comment (susanchen3) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/claysample/ClaySamplePortlet.path:56]: Also, I'm wondering what the `${key_index}` is for? I tested the path and it seems to be targeting two elements. If we want to target only **Selectable Image Card** elements, then I suggest using something like `//h4[contains(text(), '${key_header}')]//following::p[contains(@title,'${key_title}')]`

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/claysample/ClaySamplePortlet.path [hunk 3]
```diff
@@ -36,6 +36,26 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>CARD_ACTIVE</td>
+	<td>//div[contains(@class,'active file-card form-check-card form-check form-check-top-left card-type-asset')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>KEBAB_DROPDOWN_CARD_OPTION</td>
+	<td>//div[contains(@id,'image-card-icon-block')]//button[contains(@class, 'dropdown-toggle component-action btn btn-monospaced btn-unstyled')]</td>
```
Comment (susanchen3) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/claysample/ClaySamplePortlet.path:51]: Lets shorten the path here too and remove `component-action btn btn-monospaced btn-unstyled` :)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayCards.testcase [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+			AssertLocation.assertPartialLocation(value1 = "#1");
+		}
+	}
+
+	@description = "LRQA-66243: Assert default state (not checked) of Selectable Image Card"
+	@priority = "3"
+	test CheckDefaultStateSelectableImageCard {
+		task ("Check default state of Selectable Image Card") {
+			AssertNotChecked.assertNotCheckedNotVisible(locator1 = "ClaySamplePortlet#CHECKBOX_SELECTABLE_IMAGE_CARD");
+		}
```
Comment (susanchen3) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayCards.testcase:97]: Forgot to add the `Then it should not be active state` step here :)

**Comment by susanchen3** [comment]
Hi @mateusralv, looks good overall :) Just some path fix suggestions. I think some of the paths target multiple elements so if you want to target only **Selectable Image Card** I suggest starting your paths with something like `//h4[contains(text(), 'Selectable Image Card')]//following::`

**Review by susanchen3 — CHANGES_REQUESTED** [review]

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayCards.testcase [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+		task ("Assert that appends the href value to current URL: #image-card-href") {
+			AssertLocation.assertPartialLocation(value1 = "#image-card-href");
+		}
+	}
+
+	@description = "LRQA-66243: Assert that appends the href value to current URL: #1 when add on Group 1 - Option 1"
+	@priority = "5"
+	test ChangingURLWhenSelectCardGroup {
+		task ("Selec the Group 1 - Option 1") {
```
Comment (susanchen3) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClayCards.testcase:81]: Oh, also missing 't' in Select

**Comment by mateusralv** [comment]
ci:test:frontend:nocompile

**Review by john-co — APPROVED** [review]

**Comment by john-co** [comment]
ci:forward

**Pull request closed without merging**